src/components/Settings/Settings.js

```diff
@@ -0,0 +1,198 @@
+  const [authorizedApps, setAuthorizedApps] = useState( [] );
+  const [providerAuthorizations, setProviderAuthorizations] = useState( [] );
+
+  useEffect( () => {
+    let isCurrent = true;
+
+    if ( isCurrent ) {
+      getAPIToken( true ).then( setAccessToken );
+    }
+
+    return ( ) => {
+      isCurrent = false;
+    };
+  }, [] );
+
+
+  const fetchProfile = useCallback( async () => {
+    try {
+      const response = await inatjs.users.me( {api_token: accessToken} );
+      console.log( "User object", response.results[0] );
+      setSettings( response.results[0] );
+      setIsLoading( false );
+    } catch ( exc ) {
+      console.error( "exc", exc );
+      console.error( JSON.stringify( exc ) );
+    }
+  }, [accessToken] );
+
+
+  useEffect( () => {
+    async function fetchApplications() {
+      try {
+        const apps = await inatjs.authorized_applications.search( {}, {api_token: accessToken} );
+        console.log( "Authorized Applications", apps.results );
+        setAuthorizedApps( apps.results );
+        const providers = await inatjs.provider_authorizations.search( {}, {api_token: accessToken} );
+        console.log( "Provider Authorizations", providers.results );
+        setProviderAuthorizations( providers.results );
+      } catch ( exc ) {
+        console.error( "exc", exc );
+        console.error( JSON.stringify( exc ) );
+      }
+    }
+
+    if ( accessToken ) {
+      fetchProfile();
+      fetchApplications();
+    }
+  }, [accessToken, fetchProfile] );
+
+  const saveSettings = async () => {
+    setIsSaving( true );
+    const payload = {
+      id: settings.id
+    };
+    SETTINGS_PROPERTIES_LIST.forEach( ( v ) => {
+      payload[`user[${v}]`] = settings[v];
+    } );
+
+    if ( settings.removeProfilePhoto ) {
+      payload.icon_delete = true;
+    }
+    if ( settings.newProfilePhoto ) {
+      payload["user[icon]"] = {
+        type: "custom",
+        value: {
+          uri: settings.newProfilePhoto.uri,
+          type: settings.newProfilePhoto.type,
+          name: settings.newProfilePhoto.fileName
+        }
+      };
+    }
+
+    console.log( "Payload", payload );
+    const response = await inatjs.users.update( payload, {api_token: accessToken} );
+    console.log( "Updated user", response );
+    const userResponse = await inatjs.users.me( {api_token: accessToken} );
+    console.log( "User object", userResponse.results[0] );
+    setSettings( userResponse.results[0] );
+    setIsSaving( false );
+  };
+
+  const revokeApp = async ( appId ) => {
+    const response = await inatjs.authorized_applications.delete( { id: appId }, {api_token: accessToken} );
+    console.log( "Revoked app", response );
+    const apps = await inatjs.authorized_applications.search( {}, {api_token: accessToken} );
+    console.log( "Authorized Applications", apps.results );
+    setAuthorizedApps( apps.results );
+  };
+
+  return (
+    <SafeAreaView style={viewStyles.container}>
```

Comment:
Can we wrap this component in `SharedComponents/ViewWithFooter.js` so a user can get back to the drawer menu?

Reply:
Fixed in latest commit